src/util/include/util/PointerContainerView.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Xournal++
+ *
+ * A C++20-style view for containers with values pointer
+ * Such a view is not allowed to modify the pointed objects.
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <type_traits>
+
+template <typename container_type>
+class PointerContainerView {
```

Comment:
Should **be** namespaced, at least with `xoj::`.
Also, isn't this just like [std::subrange](https://en.cppreference.com/w/cpp/ranges/subrange)?
And I would use the container's iterator when possible, instead of wrapping it. That will retain the properties of the original iterators.